File: bug30_wrong_operator_precedence/solved_main.py
import base64, struct

def run():
    vals = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
    acc = 0
    for i, v in enumerate(vals):
        shifted = (acc ^ v) << (i + 1)   # FIX: was "acc ^ v << i + 1" — shift binds tighter than XOR
                                          # must add parentheses: (acc ^ v) << (i + 1)
        acc = (shifted * 0x9E3779B97F4A7C15) & 0xFFFFFFFFFFFFFFFF
        acc = ((acc << 7) | (acc >> 57)) & 0xFFFFFFFFFFFFFFFF
    raw = struct.pack(">Q", acc)
    print(base64.b64encode(raw).decode())

run()

File: bug30_wrong_operator_precedence/solved_main.js
function run() {
    let vals=[2n,3n,5n,7n,11n,13n,17n,19n,23n,29n];
    let acc=0n;
    for(let i=0n;i<10n;i++){
        let v=vals[Number(i)];
        // FIX: was "v << (i+1n)" then XOR with acc — shift applied before XOR
        // Must do XOR first: (acc ^ v) << (i+1n)
        let shifted=BigInt.asUintN(64,(acc^v)<<(i+1n));
        acc=BigInt.asUintN(64,shifted*0x9E3779B97F4A7C15n);
        acc=BigInt.asUintN(64,(acc<<7n)|(acc>>57n));
    }
    let buf=Buffer.alloc(8);
    buf.writeBigUInt64BE(acc,0);
    console.log(buf.toString('base64'));
}
run();
